packages/web-app-files/tests/unit/helpers/collaborators.spec.js

```diff
@@ -0,0 +1,89 @@
+import {
+  permissionsBitmask,
+  roleToBitmask,
+  bitmaskToRole,
+  checkPermission
+} from '../../../src/helpers/collaborators.js'
+
+describe('collaborators', () => {
+  it('permissionBitmask', () => {
+    expect(permissionsBitmask).toMatchObject({
+      read: 1,
+      update: 2,
+      create: 4,
+      delete: 8,
+      share: 16,
+      all: 31
+    })
+  })
+
+  describe('roleToBitmask function', () => {
+    describe('if additionalPermissions', () => {
+      it('should return the sum of role permissions and additional permissions bitmasks if the given role has provided additional permission', () => {
+        expect(
+          roleToBitmask(
+            {
+              additionalPermissions: {
+                share: true
+              },
+              permissions: ['read']
+            },
+            ['share', 'update']
+          )
+        ).toBe(17)
+      })
+    })
+    it('should return the sum of role permission bitmask if no additionalPermissions are given', () => {
+      expect(
+        // role additional permission is ignored if additionalPermissions is empty
+        roleToBitmask(
+          {
+            additionalPermissions: { share: true },
+            permissions: ['read', 'create']
+          },
+          []
+        )
+      ).toBe(5)
+    })
+  })
+
+  describe('bitmaskToRole function', () => {
+    it('should return advancedRole if rolePermissionsBitmask or roleBasicPermissionsBitmask does not equal provided bitmask', () => {
```

Comment:
this description and the comment bellow does refer to internal variable names of the function, we should not do that, those things might be refactored. We only want to test and rely on the external API of the function, this applies also to the description, otherwise if those variables will be refactored in the function the test might still pass, but neither the description nor the comment would make any sense.

Also more variations of the test are needed. What about if its not a folder, different bitmask, invalid values, etc.